jormungandr/src/fragment/process.rs

```diff
@@ -59,11 +60,11 @@ impl Process {
     ) -> Result<(), Error> {
         async fn hourly_wakeup(enabled: bool) {
             if enabled {
-                let now = Utc::now();
-                let current_hour = now.duration_trunc(Duration::hours(1)).unwrap();
-                let next_hour = current_hour + Duration::hours(1);
-                let sleep_duration = (next_hour - now).to_std().unwrap();
-                tokio::time::sleep(sleep_duration).await
+                let now = OffsetDateTime::now_utc();
+                // truncate date to hour so that rotation always happens at the hour mark
+                let current_hour = now.replace_time(Time::from_hms(now.hour(), 0, 0).unwrap());
+                let next_hour = current_hour + Duration::HOUR;
+                tokio::time::sleep((next_hour - now).try_into().unwrap()).await
             } else {
                 future::pending().await
```

Comment:
unrelated, but if the decision to do something comes as an argument I wonder why not just avoid calling the function in the first place

Reply:
I guess it's to avoid loads of branching points in the same place and improve readability